### ISSUE TITLE
Unbound should reply to CARP-addresses when "Network Interfaces" are configured

### ISSUE THREAD

**Issue by jPhy** [opnsense/core#5585]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Until version 21.7.7, DNS queries sent to the unbound service via either the interface's IPv4-Address or a virtual IPv4 CARP-address on the same interface were answered whenever the corresponding interface was configured as a "Network Interfaces" in unbound.
Since version 21.1, unbound does not answer queries directed to a CARP IP-address when the corresponding interface is configured as "Network Interface". 

*Tip*: to validate your setup was working with the previous version, use opnsense-revert (https://docs.opnsense.org/manual/opnsense_tools.html#opnsense-revert)

**To Reproduce**

1. Go to the "Interfaces: Virtual IPs: Settings" Page and set up virtual IP with Mode CARP-Mode for your LAN interface.
2. Go to the "Services: Unbound DNS: General" Page, enable unbound and set "Network Interfaces" to LAN
3. Open a terminal (e.g. ssh) session to the OPNSense server and run "nslookup github.com \<your CARP-Address\>"
4. Observe the error message ";; connection timed out; no servers could be reached"

**Expected behavior**

In step 4 above, the DNS-query should be answered; i.e. a CARP-Address associated with an interface picked in the "Network Interfaces" should be listened on. This was the behavior until version 21.7.7.
Alternatively, the CARP-Address could appear as a separate entry in the dropdown menu "Network Interfaces".

**Describe alternatives you considered**

As a workaround for now, I do not configure "Network Interfaces" which makes unbound listen on all interfaces including the CARP-address. That workaround however gives rise to the issue described in #5355.

**Screenshots**

![image](https://user-images.githubusercontent.com/5910377/154779951-a66e4ee0-7c9f-4ea9-8d50-6117fcfc3406.png)

**Relevant log files**

The query to the CARP-Address is not logged at all in unbound.

**Additional context**

The firewall rules must allow incoming Packets on LAN Port 53 (or whatever you configure unbound to listen on) of course.

**Environment**

OPNsense 22.1.1_3-amd64

**Comment by fichtner** [comment]
First and foremost you should base your report on the contents of /var/unbound/unbound.conf contents. If you test locally make sure no policy based routing interferes with you rest. It’s better to test from a client in this case. Lastly, maybe you set no-bind option in CARP VIP which is available since 22.1.1?

**Comment by jPhy** [comment]
> maybe you set no-bind option in CARP VIP which is available since 22.1.1?

**Indeed. That solves the issue.** The "Allow service binding" checkbox was unchecked and I never touched that setting. Maybe that new checkbox should be ticked by default for VIPs created in earlier versions (without the no-binding option) to prevent a behavioral change? It was definitely not clear to me that I'd have to adapt my configuration from the release notes.

**Comment by fichtner** [comment]
I’m not sure. We should have fixed the default in 22.1.1 along with the CARP support addition. But the question is how you ended up with no-bind in CARP if 22.1 didn’t support setting it? Or did you mean an IP alias with a VHID set? That would at least make sense then. 🤔

**Comment by jPhy** [comment]
You say the feature was added in 22.1.1 which, according to https://forum.opnsense.org/index.php?topic=26992.0, was released on February 16?

However, I could restore a configuration from "SYSTEM: CONFIGURATION: HISTORY" timestamped "2/12/22 01:38:22" with the nobind-tag on my carp configuration. The relevant section of the config is (password and subnet removed):

```
<virtualip>
    <vip>
      <type>single</type>
      <subnet_bits>32</subnet_bits>
      <mode>carp</mode>
      <interface>lan</interface>
      <subnet>192.168.*.*</subnet>
      <vhid>1</vhid>
      <advskew>110</advskew>
      <advbase>1</advbase>
      <password>..../password>
      <nobind>1</nobind>
    </vip>
  </virtualip>
```

Does that mean I already had the `<nobind>1</nobind>` in my configuration before or was supported? Very strange. I'm happy to help you further investigate if desired.

**Comment by fichtner** [comment]
@jPhy due to boolean inversion inside config.xml it did save "nobind" enabled (bind disabled) option for all types not supporting it, which included CARP in 22.1 specifically. The patch takes care of this preventing it for all types not IP alias/CARP as supported by 22.1.1 but it's really only cosmetics at this point since 22.1 can't be fixed retrospectively.


Cheers,
Franco
